### ISSUE TITLE
Action System difficulties

### ISSUE THREAD

**Issue by permcody** [idaholab/moose#4927]
Users must be careful when registering tasks to satisfy required tasks. The Action system is designed so that "tasks" can be marked as required but not individual Actions themselves. If you have an Action that you assume will always be run because it is associated with a required task, that assumption may fail if someone else later tacks on another Action to that task. If the other Action appears than the first one will no longer be created because the task will have already been satisfied.

Possible solutions might involve marking an Action as required in concert with tasks that are required.

**Comment by friedmud** [comment]
I don't like the idea of required Actions.  The whole point of the system is that it shouldn't matter how a Task gets satisfied, as long as it does.  That allows users to replace Actuons easily... as long as they satisfy the Task.

If there is a "side effect" of an Action that is required even if the Task can be satisfied without doing that... then that means that that Task should be split into two... with two Actions each independently satisfying their tasks...

**Comment by permcody** [comment]
I agree @friedmud. In fact we Andrew and I agreed that we'd split the task in the case where this popped up. The only real issue with the current system is the possibility of silently dropping one of your required "Actions" by accidentally creating another Action that satisfies the same task. Yes, that's user error but it's not easily detectable. I doubt we'll do anything about this in the short term. If it pops up again we could always create some debugging option that helps find these kinds of cases. Changing priority to minor.
